apps/alert_processor/lib/service_info/service_info_cache.ex

```diff
@@ -652,14 +658,31 @@ defmodule AlertProcessor.ServiceInfoCache do
   end
 
   defp fetch_route_branches("Red") do
-    {:ok, route_shapes} = ApiClient.route_shapes("Red")
+    {:ok, patterns, included} = ApiClient.route_patterns_with_stops("Red")
 
-    route_shapes
-    |> Enum.reject(fn %{"attributes" => %{"priority" => priority}} ->
-      priority == -1
+    included_by_type =
+      included
+      |> Enum.group_by(fn %{"type" => ty} -> ty end)
+
+    stops_by_id =
+      included_by_type["stop"]
+      |> Map.new(fn stop -> {stop["id"], stop} end)
+
+    trips_by_id =
+      included_by_type["trip"]
+      |> Map.new(fn trip -> {trip["id"], trip} end)
+
+    patterns
+    |> Enum.filter(fn %{"attributes" => %{"typicality" => t}} -> t == 1 end)
+    |> Enum.map(fn %{"relationships" => %{"representative_trip" => %{"data" => %{"id" => id}}}} ->
+      Map.get(trips_by_id, id)
     end)
-    |> Enum.map(fn %{"relationships" => %{"stops" => %{"data" => stops}}} ->
-      Enum.map(stops, & &1["id"])
+    |> Enum.map(fn %{"relationships" => %{"stops" => %{"data" => data}}} ->
+      data
+      |> Enum.map(fn %{"id" => id} -> Map.get(stops_by_id, id) end)
```

Comment:
potential issue: These uses of `Map.get` could instead be `Map.fetch!`, which crashes if the key isn't there. It would be surprising if it wasn't, since it would mean the API returned an internally-inconsistent response (or we made a mistake in our code); and in this case we'd probably want to fail rather than try to continue with some unexpected `nil`s thrown in.